MovieChecker/MovieChecker/BoxOfficeViewController.m

```diff
@@ -0,0 +1,148 @@
+                                     urlPath:movieData[@"links"][@"self"]];
+        movie.rating = movieData[@"mpaa_rating"];
+        movie.synopsis = movieData[@"synopsis"];
+        movie.year = [movieData[@"year"] intValue];
+        movie.runtime = [movieData[@"runtime"] intValue];
+        [self.movies addObject:movie];
+    }
+    [self.tableView reloadData];
+}
+
+- (void)imageForMovie:(Movie *)movie forCell:(BoxOfficeCell *)cell atIndexPath:(NSIndexPath *)indexPath
+{
+    NSURL *url = [NSURL URLWithString:movie.imagePath];
+    NSURLSessionDataTask *downloadImageTask = [[NSURLSession sharedSession]
+                                                   dataTaskWithURL:url
+                                                   completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                                       if (!error) {
+                                                           NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+                                                           if (httpResp.statusCode == 200) {
+                                                               UIImage *downloadedImage = [[UIImage alloc] initWithData:data];
+                                                               //DLOG(@"");
+                                                               movie.image = downloadedImage;
+                                                               dispatch_async(dispatch_get_main_queue(), ^{
+                                                                   if (cell.tag == indexPath.row) {
+                                                                       cell.thumbnailImageView.image = movie.image;
+                                                                   }
+                                                               });
+                                                           } else {
+                                                               // TODO: Handle the bad status code!
+                                                           }
+                                                       } else {
+                                                           // TODO: Handle the error!
+                                                       }
+                                                   }];
+    [downloadImageTask resume];
+}
+
+#pragma mark - UITableViewDataSource methods
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return self.movies.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    // TODO: Fix Image Loading to not be laggy.
+    BoxOfficeCell *boxOfficeCell = (BoxOfficeCell *)[tableView dequeueReusableCellWithIdentifier:@"BoxOfficeMovieCell" forIndexPath:indexPath];
```

Comment:
It's best to avoid inline string literals for things like this (search, e.g., "stringly typed code").  If you make your identifier the same as the cell class name, you can use `NSStringFromClass()` to get the identifier without having the string literal.